### PR TITLE
Update usage of pytest.raises throughout the codebase

### LINKED ISSUE #1541: Update use of pytest.raises to include match keyword
We should update the common pattern:

```python
with pytest.raises(Exception) as exc:
    thing()
assert str(exc.value) == message
```

to use the supported `match` keyword for pytest.raises, see https://docs.pytest.org/en/7.1.x/how-to/assert.html#assertions-about-expected-exceptions.

The pattern would then be

```python
with pytest.raises(Exception, match=message):
    thing()
```

### PR DESCRIPTION
**Pull request by duncanmmacleod**
This PR closes #1541 by updating the use of `pytest.raises` throughout the codebase to use the `match` keyword. This cleans things up a wee bit, and naturally allows more flexibility in matching the error message (since pytest uses a regex, rather than simple string comparisons).

### CODE REVIEW

**Comment by duncanmmacleod** [comment]
Ping @gwpy/maintainers, please review this change.

File: gwpy/timeseries/tests/test_core.py [hunk 1]
```diff
@@ -269,9 +278,13 @@ def test_span(self, instance):
         assert instance.span == reduce(
             operator.or_, (ts.span for ts in instance.values()), Segment(0, 0),
         )
-        with pytest.raises(ValueError) as exc:
+
+    def test_span_error_empty(self):
```
Comment (js01) [inline gwpy/timeseries/tests/test_core.py:282]: could write docstrings for tests, at least for new ones

**Review by js01 — APPROVED** [review]
From what I can see, the rewritten pytest.raises look correct

For any new tests added, could be worth adding docstrings if desired

**Comment by js01** [comment]
Codecov fails -- if I read that report correctly, coverage % is checked in the test files themselves

**Pull request merged**